packages/jellyfish-api-core/__tests__/category/account/sendTokensToAddress.test.ts

```diff
@@ -0,0 +1,163 @@
+
+    const addrReceiver = await client.wallet.getNewAddress()
+    const hex = await client.account.sendTokensToAddress({}, { [addrReceiver]: ['2@CAT'] }, {
+      selectionMode: SelectionModeType.CRUMBS
+    })
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const accCrumbsAfter = (await client.account.getAccount(addrCCrumbs))[0]
+    expect(accToNum(accCrumbsAfter)).toStrictEqual(accToNum(accCrumbsBefore) - 2)
+
+    const accReceiver = (await client.account.getAccount(addrReceiver))[0]
+    expect(accReceiver).toStrictEqual('2.00000000@CAT')
+  })
+
+  // NOTE(canonrother): "selectionMode: forward" picks the address name order by ASC, its hard to test as address is random generated
+  it.skip('should sendTokensToAddress with selectionMode forward', async () => {
+    const accForwardBefore = (await client.account.getAccount(addrCCrumbs))[0]
+
+    const addrReceiver = await client.wallet.getNewAddress()
+    const hex = await client.account.sendTokensToAddress({}, { [addrReceiver]: ['6@CAT'] }, {
+      selectionMode: SelectionModeType.FORWARD
+    })
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const accForwardAfter = (await client.account.getAccount(addrCCrumbs))[0]
+    expect(accToNum(accForwardAfter)).toStrictEqual(accToNum(accForwardBefore) - 6)
+
+    const accReceiver = (await client.account.getAccount(addrReceiver))[0]
+    expect(accReceiver).toStrictEqual('6.00000000@CAT')
+  })
+
+  it('should sendTokensToAddress with multiple receiver tokens', async () => {
+    const address = await client.wallet.getNewAddress()
+    const hex = await client.account.sendTokensToAddress({}, { [address]: ['2@ANT', '5@CAT', '10@BAT'] })
+
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    expect(await client.account.getAccount(address)).toStrictEqual(['2.00000000@ANT', '10.00000000@BAT', '5.00000000@CAT'])
+  })
+
+  it('should sendTokensToAddress with source address', async () => {
+    const addrReceiver = await client.wallet.getNewAddress()
+    const hex = await client.account.sendTokensToAddress({ [addrA]: ['10@ANT'] }, { [addrReceiver]: ['10@ANT'] })
+
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    expect(await client.account.getAccount(addrReceiver)).toStrictEqual(['10.00000000@ANT'])
+  })
+
+  it('should fail and throw an exception when no receiver address', async () => {
+    await expect(client.account.sendTokensToAddress({}, {})).rejects.toThrow('zero amounts in "to" param')
+  })
+
+  it('should throw an error when insufficient fund', async () => {
+    const promise = client.account.sendTokensToAddress({}, { [await client.wallet.getNewAddress()]: ['500@CAT'] })
```

Comment:
Choose either `container.call('getnewaddress')` or `await client.wallet.getNewAddress()` and use it in your code.  I would use `container.call('getnewaddress')`

Reply:
Refactored. From henceforth my code will be consistent. Thanks for getting sense into me.